holoviews/core/spaces.py

```diff
@@ -513,6 +515,25 @@ def __call__(self, *args, **kwargs):
         return ret
 
 
+
+class Generator(Callable):
+    """
+    Generators are considered a special case of Callable that accept no
+    arguments and never memoize.
+    """
+
+    callable = param.Parameter(default=None, constant=True, doc="""
+         The generator function being wrapped.""")
+
+    @property
+    def argspec(self):
+        from inspect import ArgSpec
+        return ArgSpec(args=[], varargs=None, keywords=None, defaults=None)
```

Comment:
Same (empty) signature as for functions with no arguments. Not really used but worth keeping the API consistent. I'll move the inline import to the top of the file.